inspirehep/modules/records/validators/helpers.py

```diff
@@ -0,0 +1,77 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Helpers for validators module."""
+
+from invenio_db import db
+from jsonschema import ValidationError
+from collections import deque
+
+FIELD_REQUIRE_FIELD_VALUES_TEMPLATE = "'{field}' field requires '{required}' field to have " \
+                                      "at least one of the values {values}."
+FIELD_DUPLICATE_VALUES_FOUND_TEMPLATE = "'{field}' field with value '{value}' found in more than one record."
+GET_RESULTS_FOR_FIELD_PROPERTY_QUERYSTRING_TEMPLATE = "SELECT id FROM records_metadata as r, " \
+                                                      "json_array_elements(r.json -> '{field}') " \
+                                                      "as elem WHERE elem ->> '{prop}'='{value}'"
+FIELD_VALIDATION_TEMPLATE = "Field '{field}' with value '{value}' is not valid."
+
+
+def check_field_values_not_in_required_values_for_record(record, field, required_values):
+    document_type_values = record.get(field, [])
```

Comment:
This is too specific of document_type. Also assumes that the field will be a list of primitives (no objects), so not much reusable.